libkbfs/block_journal.go

```diff
@@ -72,18 +59,20 @@ type blockJournal struct {
 	log      logger.Logger
 	deferLog logger.Logger
 
-	j    diskJournal
-	refs map[BlockID]blockRefMap
+	// j is the main journal.
+	j diskJournal
 
+	// saveUntilMDFlush, when non-nil, prevents garbage collection
+	// of blocks. When removed, all the referenced blocks are
+	// garbage-collected.
+	//
+	// TODO: We only really need to save a list of IDs, and not a
```

Comment:
Hmm, yeah I see how that could happen.  I think you're right, that's a bug and we should append to both.  (Either that or check whether a block exists in `saveUntilMDFlush` when deciding whether to GC it or not on a flush.)

Want to do it as part of this PR, or should I do it separately?

As far as the "full journal" question, is there another pre-existing data structure that stores a list of block IDs on disk and lets you iterate on them?  Maybe it's overkill but it was the quickest path to success and doesn't seem wasteful to me.